### PR TITLE
Approval util functions

### PR DESCRIPTION
**Pull request by casweeney**
## Description

- Implemented `approveTokens()` function.
- Implemented `getApprovedAmount()` function.

## Related Issue(s)

Closes: https://github.com/BlockheaderWeb3-Community/autoswappr-dapp/issues/39
## Checklist:

- [-] Read the [contributing docs](../CONTRIBUTING.md) (if this is your first contribution)
- [-] Verified this is not a duplicate of any existing pull request

- [-] Performed a self-review of the code
- [-] Rebased to the last commit of the target branch (or merged it into my branch)
- [-] Linked the issues which this PR resolves

### CODE REVIEW

File: app/utils/helper.ts
Comment (Utilitycoder) [inline app/utils/helper.ts:1]: Why don't we use a uniform type for both contractAddress and spender?
```typescript
(contractAddress: `0x${string}`, spender: 0x${string}`)
```

File: pnpm-lock.yaml
Comment (Utilitycoder) [inline pnpm-lock.yaml:1]: There's a merge conflict here. Please resolve the conflict.

File: app/utils/helper.ts
Comment (casweeney) [reply app/utils/helper.ts:1]: > Why don't we use a uniform type for both contractAddress and spender?
> 
> ```ts
> (contractAddress: `0x${string}`, spender: 0x${string}`)
> ```

Resolved

File: pnpm-lock.yaml
Comment (casweeney) [reply pnpm-lock.yaml:1]: > There's a merge conflict here. Please resolve the conflict.

Merge conflict fixed.

**Pull request merged**